lib/news_scraper/extractors/google_news_rss.rb

```diff
@@ -39,18 +37,18 @@ def google_urls_from_resp(body)
         end
       end
 
-      def extract_article_uris(google_urls)
+      def extract_article_urls(google_urls)
         google_urls.map do |google_url|
-          regex = google_url.match(/&url=https?:\/\/(.*)/)
-          regex.nil? ? nil : regex[1]
+          regex = google_url.match(%r(&url=(?<url>https?://.*)))
+          regex.nil? ? nil : regex['url']
         end.compact.uniq
       end
 
-      def write_to_temp(article_uris)
-        filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
-        puts "Writing article uris to #{filename}"
-        File.open(File.join(TMP_DIR, filename), 'w') do |file|
-          article_uris.each { |uri| file.puts uri }
+      def write_to_temp(article_urls)
+        filename = "#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
+        puts "Writing article urls to #{filename}"
+        File.open(File.join(ARTICLE_URLS_DIR, filename), 'w') do |file|
```

Comment:
Initially I was worried that querying the RSS feed too many times would
blacklist me so I decided to cache the urls just in case it happened.

This can be removed by the first release.

On Sep 1, 2016 8:15 PM, "Julian Nadeau" <notifications@github.com> wrote:
>
> In lib/news_scraper/extractors/google_news_rss.rb:
>
> > end.compact.uniq > end > > - def write_to_temp(article_uris) > -
filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt" > -
puts "Writing article uris to #{filename}" > - File.open(File.join(TMP_DIR,
filename), 'w') do |file| > - article_uris.each { |uri| file.puts uri } > +
def write_to_temp(article_urls) > + filename =
"#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt" > + puts "Writing
article urls to #{filename}" > + File.open(File.join(ARTICLE_URLS_DIR,
filename), 'w') do |file|
>
> We don't need any of this tmp dir stuff IMHO.
>
> Dir.mktmpdir do |dir| filename = "#{@query.downcase.gsub(/\s/,
'_')}_#{Time.now.to_i}.txt" File.open(File.join(dir, filename), 'w') do
|file| article_urls.each { |url| file.puts url } end end
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.